test/unit/connectors/GoogleAnalyticsSenderConnectorSpec.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.connectors
+
+import org.mockito.ArgumentMatchers.{eq => meq, _}
+import org.mockito.Mockito._
+import org.mockito.{ArgumentCaptor, ArgumentMatchers}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import play.api.Configuration
+import play.api.http.HeaderNames._
+import play.api.http.MimeTypes
+import play.api.libs.json.{JsValue, Json, Writes}
+import uk.gov.hmrc.customs.api.common.config.{ServiceConfig, ServiceConfigProvider}
+import uk.gov.hmrc.customs.api.common.logging.CdsLogger
+import uk.gov.hmrc.customs.notification.connectors.GoogleAnalyticsSenderConnector
+import uk.gov.hmrc.customs.notification.controllers.CustomHeaderNames.{X_CDS_CLIENT_ID_HEADER_NAME, X_CONVERSATION_ID_HEADER_NAME}
+import uk.gov.hmrc.customs.notification.logging.NotificationLogger
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.play.test.UnitSpec
+import util.MockitoPassByNameHelper.PassByNameVerifier
+import util.TestData._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GoogleAnalyticsSenderConnectorSpec extends UnitSpec with MockitoSugar with BeforeAndAfterEach {
+
+  private val mockHttpClient = mock[HttpClient]
+  private val mockCdsLogger = mock[CdsLogger]
```

Comment:
wrong code. `private val notificationLogger = new NotificationLogger(mockCdsLogger)` should be `private val notificationLogger = new NotificationLogger(NotificationLogger)` . This will also fix your problem with verifying logging.

Reply:
👍